CONTRIBUTING.md

```diff
@@ -2,32 +2,97 @@
+
+```sh
+# Show what changes git detects.
+# Not essential but good practice as it may alert you to an issue.
+git status
+
+# Add the file you've created
+git add people/username.md
+```
 
-***
+Next you will have to make a commit, which takes a snapshot of everything that's been staged (using `git add` above). To commit, run `git commit`. A text editor should appear that allows you to write a [commit message](https://chris.beams.io/posts/git-commit/) where you should describe the changes in the commit. If all succeeds, you will have added to the git repository and your commit will be visible when you run `git log`.
 
-_Note: You do not have to read the following information to contribute._
+Next you will push your commit to GitHub, which means add it to `username/sandbox` on GitHub. To do this, run:
 
-## Commit Access
+```sh
+# This will only have to be run once per computer
+# It adopts the Git 2.0, removing a confusing message
+git config --global push.default simple
+
+# Push your current branch to GitHub
+# If this is the first time you're pushing this branch,
+# you will receive an error suggesting you specify --set-upstream.
+# If so, use the suggested command. Then future git push attempts will work.
+git push
+```
+
+Now you are ready to create your [pull request](https://help.github.com/articles/about-pull-requests/). The GitHub webpage for either `cognoma/sandbox` or `username/sandbox` should have a new button to "Compare & pull request." Click it and write a title and description for your requested changes. Then submit. Soon (hopefully), another Cognoma contributor will review your pull request. Usually, pull requests will require some changes before they are ready to merge (incorporate into Cognoma repository). You can add additional changes to a pull request by adding commits to the branch its based on, which is this example is `add-username`.
+
+Once your pull request passes review, it will be merged by a repository maintainer. Before starting on any new pull requests, you will want to sync your fork, so you're not starting from an outdated version of the codebase, by running:
+
+```
+# Fetch all new changes on the cognoma repo
+git fetch upstream
+
+# Make sure you're on your master branch.
+git checkout master
+
+# Fast forward your master branch with upstream changes.
+git merge --no-ff upstream/master
+```
+
+### Pull request advice
+
+1. Focus a pull request on a single feature. Smaller pull requests are easier to review and will be thus get merged quicker. If you want to change multiple things, create multiple pull requests.
++ Open incomplete pull requests but put WIP in the title for "work in progress". This enables early stage feedback to save time later and helps coordinate efforts. Once you feel the pull request is ready, remove WIP from the title.
```

Comment:
This didn't seem to work, which surprises me as it's pretty standard markdown. Could be because I added line breaks. But anyways I switched to explicit numbering everywhere.